frontend/ui/elements/screensaver_menu.lua

```diff
@@ -26,114 +26,156 @@ local function genMenuItem(text, setting, value, enabled_func, separator)
+            genMenuItem(_("Show custom image on sleep screen"), "screensaver_type", "image_file"),
+            genMenuItem(_("Use random image from folder on sleep screen"), "screensaver_type", "random_image"),
+            genMenuItem(_("Show document cover on sleep screen"), "screensaver_type", "document_cover"),
+            genMenuItem(_("Show reading progress on sleep screen"), "screensaver_type", "readingprogress", isReaderProgressEnabled),
+            genMenuItem(_("Show book status on sleep screen"), "screensaver_type", "bookstatus", hasLastFile),
+            genMenuItem(_("Leave screen as-is"), "screensaver_type", "disable", nil, true),
+            separator = true,
             {
-                text = _("Screensaver folder"),
-                keep_menu_open = true,
-                callback = function()
-                    Screensaver:chooseFolder()
+                text = _("Border fill"),
+                enabled_func = function()
+                    return G_reader_settings:readSetting("screensaver_type") == "cover"
+                            or G_reader_settings:readSetting("screensaver_type") == "random_image"
+                            or G_reader_settings:readSetting("screensaver_type") == "image_file"
+                            or G_reader_settings:readSetting("screensaver_type") == "document_cover"
                 end,
+                sub_item_table = {
+                    genMenuItem(_("Black"), "screensaver_img_background", "black"),
+                    genMenuItem(_("White"), "screensaver_img_background", "white"),
+                    genMenuItem(_("Current state"), "screensaver_img_background", "none", nil, true),
+                    -- separator
+                        {
+                            text_func = function()
+                                local percentage = G_reader_settings:readSetting("screensaver_stretch_limit_percentage")
+                                if G_reader_settings:isTrue("screensaver_stretch_images") and percentage then
+                                    return T(_("Stretch to fit screen (with limit: %1 %)"), percentage)
+                                end
+                                return _("Stretch cover to fit screen")
+                            end,
+                            checked_func = function()
+                                return G_reader_settings:isTrue("screensaver_stretch_images")
+                            end,
+                            callback = function(touchmenu_instance)
+                                Screensaver:setStretchLimit(touchmenu_instance)
+                            end,
+                        },
+                },
             },
             {
-                text = _("Screensaver image"),
-                keep_menu_open = true,
-                callback = function()
-                    Screensaver:chooseFile()
+                text = _("Awake screen delay"),
```

Comment:
@offset-torque As `Awake screen delay` I believe this phrase would require a translator's note, which is a bit of a canary. I would change it to something like this to avoid the need for a translator's note.

```suggestion
                text = _("Screen wake-up delay"),
```

The current text is `Keep the screensaver on screen after wakeup`, which is awkward but clear.

Maybe you have a new perspective?

Reply:
**"Screen wake-up delay"** is definitely more clear than **"Awake screen delay"**. It implies that device is awake but screen is not yet, hopefully users too will understand the distinction. I think this is the best we can do if we are replacing this **"Keep the screensaver on screen after wakeup"**, it is very hard to phrase this concept.